From f9537b22e0d8d482055692bce75feb277d332950 Mon Sep 17 00:00:00 2001
From: Nitesh <kunitesh@vmware.com>
Date: Thu, 6 Jun 2024 11:47:07 +0530
Subject: [PATCH 4/4] Remove unused default groups, rules and tmpfiles

Some of these groups and users are also created by filesystem
---
 rules.d/50-udev-default.rules.in            | 11 ---------
 sysusers.d/basic.conf.in                    | 26 ---------------------
 sysusers.d/systemd-journal.conf.in          |  2 --
 sysusers.d/systemd-network.conf.in          |  2 --
 sysusers.d/systemd-remote.conf              |  2 --
 sysusers.d/systemd-resolve.conf.in          |  2 --
 sysusers.d/systemd-timesync.conf.in         |  2 --
 tmpfiles.d/static-nodes-permissions.conf.in |  3 ---
 8 files changed, 50 deletions(-)

diff --git a/rules.d/50-udev-default.rules.in b/rules.d/50-udev-default.rules.in
index 8fa518cd8f..23f43d0409 100644
--- a/rules.d/50-udev-default.rules.in
+++ b/rules.d/50-udev-default.rules.in
@@ -53,13 +53,8 @@ SUBSYSTEM=="dvb", GROUP="video"
 SUBSYSTEM=="media", GROUP="video"
 SUBSYSTEM=="cec", GROUP="video"
 
-SUBSYSTEM=="drm", KERNEL=="renderD*", GROUP="render", MODE="{{GROUP_RENDER_MODE}}"
-SUBSYSTEM=="kfd", GROUP="render", MODE="{{GROUP_RENDER_MODE}}"
 SUBSYSTEM=="accel", GROUP="render", MODE="{{GROUP_RENDER_MODE}}"
 
-SUBSYSTEM=="misc", KERNEL=="sgx_enclave", GROUP="sgx", MODE="0660"
-SUBSYSTEM=="misc", KERNEL=="sgx_vepc", GROUP="sgx", MODE="0660"
-
 # When using static_node= with non-default permissions, also update
 # tmpfiles.d/static-nodes-permissions.conf.in to keep permissions synchronized.
 
@@ -108,15 +103,9 @@ KERNEL=="tun", MODE="0666", OPTIONS+="static_node=net/tun"
 KERNEL=="fuse", MODE="0666", OPTIONS+="static_node=fuse"
 
 # The static_node is required on s390x and ppc (they are using MODULE_ALIAS)
-KERNEL=="kvm", GROUP="kvm", MODE="{{DEV_KVM_MODE}}", OPTIONS+="static_node=kvm"
 
 KERNEL=="vfio", MODE="0666", OPTIONS+="static_node=vfio/vfio"
 
 KERNEL=="vsock", MODE="0666"
-KERNEL=="vhost-vsock", GROUP="kvm", MODE="{{DEV_KVM_MODE}}", OPTIONS+="static_node=vhost-vsock"
-
-KERNEL=="vhost-net", GROUP="kvm", MODE="{{DEV_KVM_MODE}}", OPTIONS+="static_node=vhost-net"
-
-KERNEL=="udmabuf", GROUP="kvm"
 
 LABEL="default_end"
diff --git a/sysusers.d/basic.conf.in b/sysusers.d/basic.conf.in
index 992af346ca..c66181be00 100644
--- a/sysusers.d/basic.conf.in
+++ b/sysusers.d/basic.conf.in
@@ -5,37 +5,11 @@
 #  the Free Software Foundation; either version 2.1 of the License, or
 #  (at your option) any later version.
 
-# The superuser
-g root    0       -            -
-u root    0:0     "Super User" /root
-
 # The nobody user/group for NFS file systems
 g {{NOBODY_GROUP_NAME}} 65534       -            -
-u! {{NOBODY_USER_NAME }} 65534:65534 "Kernel Overflow User"     -
-
-# Administrator group: can *see* more than normal users
-g adm     {{ADM_GID    }}     -            -
-
-# Administrator group: can *do* more than normal users
-g wheel   {{WHEEL_GID  }}     -            -
 
 # Access to shared database of users on the system
 g utmp    {{UTMP_GID   }}     -            -
 
 # Physical and virtual hardware access groups
-g audio   {{AUDIO_GID  }}     -            -
-g cdrom   {{CDROM_GID  }}     -            -
-g dialout {{DIALOUT_GID}}     -            -
-g disk    {{DISK_GID   }}     -            -
-g input   {{INPUT_GID  }}     -            -
-g kmem    {{KMEM_GID   }}     -            -
-g kvm     {{KVM_GID    }}     -            -
-g lp      {{LP_GID     }}     -            -
-g render  {{RENDER_GID }}     -            -
-g sgx     {{SGX_GID    }}     -            -
-g tape    {{TAPE_GID   }}     -            -
 g tty     {{TTY_GID    }}     -            -
-g video   {{VIDEO_GID  }}     -            -
-
-# Default group for normal users
-g users   {{USERS_GID  }}     -            -
diff --git a/sysusers.d/systemd-journal.conf.in b/sysusers.d/systemd-journal.conf.in
index 61768b234e..5873bfab30 100644
--- a/sysusers.d/systemd-journal.conf.in
+++ b/sysusers.d/systemd-journal.conf.in
@@ -4,5 +4,3 @@
 #  under the terms of the GNU Lesser General Public License as published by
 #  the Free Software Foundation; either version 2.1 of the License, or
 #  (at your option) any later version.
-
-g systemd-journal {{SYSTEMD_JOURNAL_GID}} -
diff --git a/sysusers.d/systemd-network.conf.in b/sysusers.d/systemd-network.conf.in
index fc04827efd..5873bfab30 100644
--- a/sysusers.d/systemd-network.conf.in
+++ b/sysusers.d/systemd-network.conf.in
@@ -4,5 +4,3 @@
 #  under the terms of the GNU Lesser General Public License as published by
 #  the Free Software Foundation; either version 2.1 of the License, or
 #  (at your option) any later version.
-
-u! systemd-network {{SYSTEMD_NETWORK_UID}} "systemd Network Management"
diff --git a/sysusers.d/systemd-remote.conf b/sysusers.d/systemd-remote.conf
index 796850c9e6..5873bfab30 100644
--- a/sysusers.d/systemd-remote.conf
+++ b/sysusers.d/systemd-remote.conf
@@ -4,5 +4,3 @@
 #  under the terms of the GNU Lesser General Public License as published by
 #  the Free Software Foundation; either version 2.1 of the License, or
 #  (at your option) any later version.
-
-u! systemd-journal-remote  - "systemd Journal Remote"
diff --git a/sysusers.d/systemd-resolve.conf.in b/sysusers.d/systemd-resolve.conf.in
index e385070c45..5873bfab30 100644
--- a/sysusers.d/systemd-resolve.conf.in
+++ b/sysusers.d/systemd-resolve.conf.in
@@ -4,5 +4,3 @@
 #  under the terms of the GNU Lesser General Public License as published by
 #  the Free Software Foundation; either version 2.1 of the License, or
 #  (at your option) any later version.
-
-u! systemd-resolve {{SYSTEMD_RESOLVE_UID}} "systemd Resolver"
diff --git a/sysusers.d/systemd-timesync.conf.in b/sysusers.d/systemd-timesync.conf.in
index 7b9fb3d8d9..5873bfab30 100644
--- a/sysusers.d/systemd-timesync.conf.in
+++ b/sysusers.d/systemd-timesync.conf.in
@@ -4,5 +4,3 @@
 #  under the terms of the GNU Lesser General Public License as published by
 #  the Free Software Foundation; either version 2.1 of the License, or
 #  (at your option) any later version.
-
-u! systemd-timesync {{SYSTEMD_TIMESYNC_UID}} "systemd Time Synchronization"
diff --git a/tmpfiles.d/static-nodes-permissions.conf.in b/tmpfiles.d/static-nodes-permissions.conf.in
index df1fe2e3d5..d85c44c175 100644
--- a/tmpfiles.d/static-nodes-permissions.conf.in
+++ b/tmpfiles.d/static-nodes-permissions.conf.in
@@ -15,6 +15,3 @@ z /dev/loop-control 0660 - disk  -
 z /dev/net/tun      0666 - -     -
 z /dev/fuse         0666 - -     -
 z /dev/vfio/vfio    0666 - -     -
-z /dev/kvm          {{DEV_KVM_MODE}} - kvm -
-z /dev/vhost-net    {{DEV_KVM_MODE}} - kvm -
-z /dev/vhost-vsock  {{DEV_KVM_MODE}} - kvm -
-- 
2.52.0

